### ISSUE TITLE
BUG: CC0022 DisposableVariableNotDisposedAnalyzer: False positive for expression bodied members

### ISSUE THREAD

**Issue by paulomorgado** [code-cracker/code-cracker#880]
The **CC0022 DisposableVariableNotDisposedAnalyzer** is flagging a return of an `IDisposable` in an expression bodied while not on a "normal" member.

````csharp
class D : IDisposable
{
    void IDisposable.Dispose() { }
}

public class C
{
    public IDisposable M1() => new D();

    public IDisposable M2()
    {
        return new D();
    }
}
````
![image](https://cloud.githubusercontent.com/assets/470455/23402554/f47090b2-fda3-11e6-96e1-fd294a748b4f.png)

**Comment by carloscds** [comment]
@paulomorgado But bodied not implement Dispose() method automatically. right ?

**Comment by paulomorgado** [comment]
@carloscds, the point is that `M1` and `M2` are the same thing and should be treated the same.

The object is being returned and the caller should dispose of it.

Can you point me to the analyzer code and tests?

**Comment by carloscds** [comment]
@paulomorgado 
Code: https://github.com/code-cracker/code-cracker/blob/master/src/CSharp/CodeCracker/Usage/DisposableVariableNotDisposedAnalyzer.cs

Test: https://github.com/code-cracker/code-cracker/blob/master/test/CSharp/CodeCracker.Test/Usage/DisposableVariableNotDisposedTests.cs

**Comment by giggio** [comment]
@paulomorgado you are correct, it is a bug, `CC0022` should not be raised for the expression bodied method either.

**Comment by giggio** [comment]
Closed on #900
